### PR TITLE
ControllerCMDS.registerNode() now correctly updates the Daemon's Session object with the newly-assigned Node ID

### PR DESCRIPTION
**Pull request by amrhassan**

### CODE REVIEW

File: lib/JumpScale/grid/serverbase/Daemon.py [hunk 1]
```diff
@@ -16,6 +16,12 @@ def __init__(self, ddict):
     def __repr__(self):
         return str(self.__dict__)
 
+    def updateNodeId(self, new_node_id):
+        """
+        Sets the session's NID attribute to the provided new_node_id.
+        """
+        self.__dict__['nid'] = new_node_id
```
Comment (grimpy) [inline lib/JumpScale/grid/serverbase/Daemon.py]: self.nid = new_node_id

File: lib/JumpScale/grid/serverbase/Daemon.py [hunk 2]
```diff
@@ -122,6 +128,20 @@ def decrypt(self, message, session):
...
+        Notifies this daemon about a newly-registered node.
+
+        Args:
+            node: metadata about the new node.
+            session_id (str): the ID of the session the new node is involved in.
+        """
+
+        if hasattr(node, 'id'):
+            # Let's use this opportunity to update the associated session with the new NID
+            self.sessions[session_id].updateNodeId(getattr(node, 'id'))
```
Comment (grimpy) [inline lib/JumpScale/grid/serverbase/Daemon.py]: node.id

File: lib/JumpScale/grid/serverbase/Daemon.py [hunk 3]
```diff
@@ -122,6 +128,20 @@ def decrypt(self, message, session):
         else:
             return message
 
+    def notifyOfNewNode(self, node, session_id):
```
Comment (grimpy) [inline lib/JumpScale/grid/serverbase/Daemon.py]: maybe call this updateSession?

**Comment by amrhassan** [comment]
@grimpy *nudge*

File: lib/JumpScale/grid/serverbase/Daemon.py [hunk 4]
```diff
@@ -12,10 +12,17 @@ class Session():
 
     def __init__(self, ddict):
         self.__dict__ = ddict
+        self.nid = None
```
Comment (grimpy) [inline lib/JumpScale/grid/serverbase/Daemon.py]: ddcit contains nid you are overwritting nid with None

File: lib/JumpScale/grid/serverbase/Daemon.py [hunk 5]
```diff
@@ -122,6 +129,20 @@ def decrypt(self, message, session):
...
+        Args:
+            node: metadata about the new node.
+            session_id (str): the ID of the session the new node is involved in.
+        """
+
+        if hasattr(node, 'id'):
+            # Let's use this opportunity to update the associated session with the new NID
+            self.sessions[session_id].updateNodeId(node.id)
+
+
```
Comment (grimpy) [inline lib/JumpScale/grid/serverbase/Daemon.py]: remove a whitespace here

File: lib/JumpScale/grid/serverbase/Daemon.py (hunk 4 above)
Comment (amrhassan) [inline lib/JumpScale/grid/serverbase/Daemon.py]: @grimpy Fixed

File: lib/JumpScale/grid/serverbase/Daemon.py (hunk 5 above)
Comment (amrhassan) [inline lib/JumpScale/grid/serverbase/Daemon.py]: @grimpy removed

**Pull request merged**